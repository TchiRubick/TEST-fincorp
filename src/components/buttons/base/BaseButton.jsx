import React from 'react';

import './BaseButton.scss';

const BaseButton = (props) => {
  const { onClick, children, color } = props;

  return (
    <button type="button" onClick={onClick} className={`comp-button-base __${color || ''}`}>
      {children}
    </button>
  );
};

export default BaseButton;
