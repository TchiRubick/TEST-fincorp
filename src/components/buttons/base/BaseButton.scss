@import '../../style.scss';

.comp-button-base {
  color: white;
  border: none;
  padding: 0.8rem;
  border-radius: 0.4rem;

  &.__green {
    background-color: $color-green;
  }
}