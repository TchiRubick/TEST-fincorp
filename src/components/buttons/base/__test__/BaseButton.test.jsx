import React from 'react';
import ReactDom from 'react-dom';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import BaseButton from '../BaseButton';

afterEach(cleanup);

it('renders good', () => {
  const div = document.createElement('div');

  ReactDom.render(<BaseButton>test</BaseButton>, div);
});

it('as expected snapshot', () => {
  const tree = renderer.create(<BaseButton>test</BaseButton>).toJSON();
  expect(tree).toMatchSnapshot();
});
