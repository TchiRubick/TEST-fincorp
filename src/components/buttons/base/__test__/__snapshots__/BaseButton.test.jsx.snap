// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as expected snapshot 1`] = `
<button
  className="comp-button-base __"
  type="button"
>
  test
</button>
`;
